File: leetcode/454.4-sum-ii/ts/foursumcount.test.ts
import { test, expect } from "vitest"
import fourSumCount from "./foursumcount.js"

interface testcase {
    nums1: number[]
    nums2: number[]
    nums3: number[]
    nums4: number[]
    want: number
}

test("foursumcounts", () => {
    const tcs: testcase[] = [
        { nums1: [1, 2], nums2: [-2, -1], nums3: [-1, 2], nums4: [0, 2], want: 2 },
        { nums1: [0], nums2: [0], nums3: [0], nums4: [0], want: 1 },
    ]

    for (const tc of tcs) {
        expect(fourSumCount(tc.nums1, tc.nums2, tc.nums3, tc.nums4) === tc.want)
    }
})

File: leetcode/454.4-sum-ii/go/foursumcount_test.go
package foursumcount

import "testing"

type testcase struct {
	nums1 []int
	nums2 []int
	nums3 []int
	nums4 []int
	want  int
}

func TestFourSumCount(t *testing.T) {
	tcs := []testcase{
		{nums1: []int{1, 2}, nums2: []int{-2, -1}, nums3: []int{-1, 2}, nums4: []int{0, 2}, want: 2},
		{nums1: []int{0}, nums2: []int{0}, nums3: []int{0}, nums4: []int{0}, want: 1},
	}

	for _, tc := range tcs {
		t.Run("forsumcount", func(t *testing.T) {
			got := fourSumCount(tc.nums1, tc.nums2, tc.nums3, tc.nums4)
			if got != tc.want {
				t.Fatalf("forsumcount, want: %d, got: %d", tc.want, got)
			}
		})
	}
}
